src/main/kotlin/app/revanced/patches/youtube/layout/sponsorblock/resource/patch/ShortsPlaybackDetection.kt

```diff
@@ -0,0 +1,29 @@
+package app.revanced.patches.youtube.layout.sponsorblock.resource.patch
+
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.ResourceContext
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.ResourcePatch
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patches.shared.mapping.misc.patch.ResourceMappingPatch
+import app.revanced.patches.youtube.layout.sponsorblock.annotations.SponsorBlockCompatibility
+
+@Name("shorts-playback-detection")
+@SponsorBlockCompatibility
```

Comment:
Patches that are only being depended on do not front-face the end user and don't require a compatibility annotation. Also because multiple patches use them, so you cant constrain them to either compatibility annotation.

Reply:
I'm trying to refactor this into it's own integrations class, so it's patched once and no copy paste patch code for the dependents.
I will push changes soon.